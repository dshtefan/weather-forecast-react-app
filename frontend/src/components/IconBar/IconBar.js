import React from 'react';
import './IconBar.scss';
import updIcon from './svg/update.svg';
import errIcon from './svg/err.svg';

const IconBar = ({ locRequested, errorMessage }) => {
  return (
    <div className="icon-bar">
      <div className="icons">
        <img className="navIcon" src={updIcon} alt=""  onClick={() => locRequested()} />
        {errorMessage ? <img className="errIcon" src={errIcon} alt="" /> : null}
        {errorMessage ? <div className="error-text">{errorMessage}</div> : null}
      </div>
    </div>
  )
};

export default IconBar;