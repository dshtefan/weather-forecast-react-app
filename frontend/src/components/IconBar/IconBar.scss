.icon-bar {
  grid-area: ib;
  display: flex;
  min-height: 45px;
  align-items: flex-start;
}
.icons {
  min-height: 45px;
  display: flex;
  align-items: center;
}
.navIcon {
  cursor: pointer;
}
.errIcon{
  margin-left: 20px;
}
.error-text {
  margin-left: 10px;
  color: #D34848;
  font-size: 20px;
  white-space: nowrap;
}