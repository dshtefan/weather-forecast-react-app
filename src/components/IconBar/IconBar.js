import React from 'react';
import './IconBar.scss';
import updIcon from './svg/update.svg';
import errIcon from './svg/err.svg';
import {connect} from "react-redux";
import {locRequested} from "../../actions";

const IconBar = ({ locRequested, errorMessage }) => {
  return (
    <div className="icon-bar">
      <div className="icons">
        <img className="navIcon" src={updIcon} alt=""  onClick={() => locRequested()} />
        {errorMessage ? <img className="errIcon" src={errIcon} alt="" /> : null}
        {errorMessage ? <div className="error-text">{errorMessage}</div> : null}
      </div>
    </div>
  )
};

export default connect(({errorMessage}) => ({errorMessage}), { locRequested })(IconBar);