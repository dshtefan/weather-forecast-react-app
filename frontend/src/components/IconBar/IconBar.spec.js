import React from 'react';
import IconBar from "./IconBar";

describe('IconBar', () => {
  const props = {
    locRequested: () => {},
    errorMessage: 'error'
  };

  describe('ошибка', () => {
    const iconBar = shallow(<IconBar {...props} />);
    it ('отображается корректно', () => {
      expect(shallowToJson(iconBar)).toMatchSnapshot();
    });
  });

  describe('компонент без ошибки', () => {
    const nextProps = {
      ...props,
      errorMessage: null
    };
    const iconBar = shallow(<IconBar {...nextProps} />);
    it ('отображается корректно', () => {
      expect(shallowToJson(iconBar)).toMatchSnapshot();
    });
  });
});