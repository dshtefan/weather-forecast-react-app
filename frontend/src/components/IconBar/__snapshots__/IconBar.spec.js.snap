// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityListItem компонент без ошибки отображается корректно 1`] = `
<div
  className="icon-bar"
>
  <div
    className="icons"
  >
    <img
      alt=""
      className="navIcon"
      onClick={[Function]}
      src="update.svg"
    />
  </div>
</div>
`;

exports[`CityListItem ошибка отображается корректно 1`] = `
<div
  className="icon-bar"
>
  <div
    className="icons"
  >
    <img
      alt=""
      className="navIcon"
      onClick={[Function]}
      src="update.svg"
    />
    <img
      alt=""
      className="errIcon"
      src="err.svg"
    />
    <div
      className="error-text"
    >
      error
    </div>
  </div>
</div>
`;

exports[`IconBar компонент без ошибки отображается корректно 1`] = `
<div
  className="icon-bar"
>
  <div
    className="icons"
  >
    <img
      alt=""
      className="navIcon"
      onClick={[Function]}
      src="update.svg"
    />
  </div>
</div>
`;

exports[`IconBar ошибка отображается корректно 1`] = `
<div
  className="icon-bar"
>
  <div
    className="icons"
  >
    <img
      alt=""
      className="navIcon"
      onClick={[Function]}
      src="update.svg"
    />
    <img
      alt=""
      className="errIcon"
      src="err.svg"
    />
    <div
      className="error-text"
    >
      error
    </div>
  </div>
</div>
`;
